docs/src/spec/struct_types.md

```diff
@@ -0,0 +1,37 @@
+# Struct types
+
+A *struct type* is the type denoted by the name of an [`struct` item].
+A `struct` *type* is a heterogeneous product of other types, called the
+*fields* of the type.
+
+New instances of a `struct` can be constructed with a [struct expression].
+
+Struct types are either stored in storage or memory but are never stored directly on the stack.
+
+Examples:
+
+```Python
+struct Rectangle:
+  width: u256
+  length: u256
+
+contract Example:
+  # A Rectangle in storage
+  area: Rectangle
+
+  def do_something():
+    # A rectangle in memory
+    square: Rectangle = Rectangle(width=10, length=10)
+```
+
+All fields of struct types are always initialized.
+
+The memory layout of a `struct` is undefined by default to allow for compiler
+optimizations like field reordering.
```

Comment:
maybe this is a better note:

> The data layout of a `struct` is not part of its external API and may be changed between any release.

There are a couple things that influence the layout of struct fields: space efficiency and word alignment. Maybe we don't discuss either of these things in here and instead leave it to the data layout section.